### PR TITLE
Fix the `threadPoolSize` not working.

### PR DESCRIPTION
**Pull request by iteyelmp**
https://github.com/ethstorage/ethstorage-sdk/pull/17#issuecomment-2299254369
https://github.com/ethstorage/ethfs-cli/pull/15#pullrequestreview-2248497569

Fix the bug where the `threadPoolSize` parameter has no effect. The issue is caused by the parameter being passed as a string, which fails to be recognized correctly. As a result, the concurrency level defaults to the maximum, causing `-6` and `-12` to produce the same outcome.

After fixing the issue, the `-s` parameter will properly reflect the specified concurrency level, resulting in a noticeable difference in execution time. 

For example:
Download the repository from [https://github.com/vbuterin/blog](https://github.com/vbuterin/blog), and then upload the `site` folder. After uploading all the files, perform the upload operation again. This second operation will only compare all files and will not trigger an actual upload. The time taken is as follows:

`-s 15`: 2.4 minutes, 
`-s 6`: 4.7 minutes.
`default is 6`: 4.66 minutes.

### CODE REVIEW

**Comment by qzhodl** [comment]
As a good PR practice, providing clear comments is crucial. You should not assume that the reviewer has the full context of the issue (in fact, DL does not have context for this issue). Therefore, it is essential to include:

- A link to the reported issue
- An analysis of the root cause
- A description of how you tested the fix
- The results of your testing

File: src/utils/uploader.js [hunk 1]
```diff
@@ -129,11 +125,10 @@ class Uploader {
 
     // upload
     async upload(path, gasIncPct, threadPoolSize) {
-        const syncPoolSize = threadPoolSize || DEFAULT_THREAD_POOL_SIZE;
         const results = [];
         return new Promise((resolve, reject) => {
             from(recursiveFiles(path, ''))
-                .pipe(mergeMap(info => this.#upload(info, gasIncPct), syncPoolSize))
+                .pipe(mergeMap(info => this.#upload(info, gasIncPct), threadPoolSize))
```
Comment (qzhodl) [inline src/utils/uploader.js:131]: That is the reason that many teams choose typescript instead of javascript because TS can have type check for us

**Comment by qzhodl** [comment]
> [ethstorage/ethstorage-sdk#17 (comment)](https://github.com/ethstorage/ethstorage-sdk/pull/17#issuecomment-2299254369) [#15 (review)](https://github.com/ethstorage/ethfs-cli/pull/15#pullrequestreview-2248497569)
> 
> Fix the bug where the `threadPoolSize` parameter has no effect. The issue is caused by the parameter being passed as a string, which fails to be recognized correctly. As a result, the concurrency level defaults to the maximum, causing `-6` and `-12` to produce the same outcome.
> 
> After fixing the issue, the `-s` parameter will properly reflect the specified concurrency level, resulting in a noticeable difference in execution time.
> 
> For example: Download the repository from https://github.com/vbuterin/blog, and then upload the `site` folder. After uploading all the files, perform the upload operation again. This second operation will only compare all files and will not trigger an actual upload. The time taken is as follows:
> 
> `-s 15`: 2.4 minutes, `-s 6`: 4.7 minutes.

I believe your test plan should at least cover the scenario described in the reported issue: ensuring that the time cost is the same when using the default threadPoolSize versus -s 6

File: src/index.js [hunk 2]
```diff
@@ -229,12 +233,21 @@ const estimateAndUpload = async (key, domain, path, type, rpc, chainId, gasIncPc
...
-  const handler = await getWebHandler(domain, rpc, chainId, CHAIN_ID_DEFAULT);
+  const handler = await getWebHandler(domain, rpc, chainId, CHAIN_ID_DEFAULT, false);
   if (!handler.providerUrl || parseInt(handler.address) <= 0) {
     console.log(error(`ERROR: ${domain} domain doesn't exist`));
     return;
   }
 
+  if (threadPoolSize) {
+    threadPoolSize = Number(threadPoolSize);
+  } else if (handler.chainId === SEPOLIA_CHAIN_ID || handler.chainId === QUARKCHAIN_L2_TESTNET_CHAIN_ID) {
```
[the remark this answers is not in the document]
Comment (iteyelmp) [reply src/index.js:244]: No, his request limit is lower, but sepolia uses our own rpc by default.

[the remark this answers is not in the document]
Comment (qzhodl) [reply src/index.js:244]: So I would suggest to use DEFAULT_THREAD_POOL_SIZE_LOW for sepolia

**Review by qzhodl — APPROVED** [review]

**Pull request merged**